rfcs/refactor_project_structure.md

```diff
@@ -0,0 +1,89 @@
+# Summary
+[summary]: #summary
+
+This RFC describes a restructuring of the project into a library with separate modules for errors, forms,
+models and schemas.   
+
+# Motivation
+[motivation]: #motivation
+
+## Why Restructure the Project?
+This project is the website for the rustbridge project and aims to make
+organizing and creating new rustbridge workshops easy.  Right now the website
+content is displayed and a bare-bones implementation of the submission form for
+new workshops is available but doesn't actually work.  
+
+Right now adding new features or changing existing features is a messy process
+and changing or adding one feature impacts the implementation the others.  
+
+## Whats the Expected Outcome?
+The expected outcome of the proposed changes is a well structured project that
+is easy to navigate, build on and document without needing to change too much of
+whats already there.  
+
+# Detailed Explanation
+[detailed-explaination]: #detail-explaination
+
+This RFC proposes the following project structure:
+```
+src
+├── bin
+│   └── main.rs
+├── error
+│   ├── test
```

Comment:
2 things:

1. idiomatically, it should be `tests` not `test`, and not a `mod.rs` but `tests.rs`
2. i would keep the tests in a separate `tests` dir for the moment, not per "component", for simplicity's sake. we don't have that many tests right now and splitting will come with overhead that we don't need to pay at the moment i don't think